### PR TITLE
Docs: Fix internal links in 1.5.x releases

### LINKED ISSUE #10409: Broken links in Spark Writes documentation
### Apache Iceberg version

1.5.2 (latest release)

### Query engine

Spark

### Please describe the bug 🐞

I'm not an Iceberg user, just browsing the doc around. 
I've noticed that all links pointing to other pages are broken on https://iceberg.apache.org/docs/latest/spark-writes (there might be additional broken links in other pages but I haven't checked those).

Eg
> To use Iceberg in Spark, first configure [Spark catalogs](https://iceberg.apache.org/docs/latest/spark-configuration.md).

### PR DESCRIPTION
**Pull request by manuzhang**
This fixes #10409, #10249
cc @nastra, @Fokko, @bitsondatadev

### CODE REVIEW

**Comment by bitsondatadev** [comment]
@manuzhang have you tried running this locally?

**Comment by Fokko** [comment]
@bitsondatadev After this PR we need to fix this in the [docs branch](https://github.com/apache/iceberg/tree/docs), right?

**Comment by manuzhang** [comment]
@Fokko This PR is open against docs branch.

File: 1.5.2/docs/configuration.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ Iceberg catalogs support using catalog properties to configure catalog behaviors
 | clients                           | 2                  | client pool size                                       |
 | cache-enabled                     | true               | Whether to cache catalog entries |
 | cache.expiration-interval-ms      | 30000              | How long catalog entries are locally cached, in milliseconds; 0 disables caching, negative values disable expiration |
-| metrics-reporter-impl | org.apache.iceberg.metrics.LoggingMetricsReporter | Custom `MetricsReporter` implementation to use in a catalog. See the [Metrics reporting](../metrics-reporting.md) section for additional details |
+| metrics-reporter-impl | org.apache.iceberg.metrics.LoggingMetricsReporter | Custom `MetricsReporter` implementation to use in a catalog. See the [Metrics reporting](metrics-reporting.md) section for additional details |
```
[the remark this answers is not in the document]
Comment (Fokko) [reply 1.5.2/docs/configuration.md:136]: > At that time we most likely just haven't had backports for the link fixes

Exactly. #9965 got merged (March 23, 2024) after the 1.5.0 release (March 11, 2024), so those fixes are not on the 1.5.x branch.

[the remark this answers is not in the document]
Comment (bitsondatadev) [reply 1.5.2/docs/configuration.md:136]: Just so it's clear I still have a link fix en route, I just need to recall the decision we made regarding absolute vs relative links.

**Comment by manuzhang** [comment]
@nastra As @Fokko said, #9965 wan't merged in time for 1.5.x branch and this PR basically back-ports the patch (resolving minor conflicts). I've built site locally and manually checked some links. They are working correctly now.

**Review by Fokko — APPROVED** [review]
Looks good! Thanks for taking care of this @manuzhang 

We can check afterwards if we missed anything by running the link checker https://github.com/apache/iceberg/pull/10404#issuecomment-2146176153

**Review by nastra — APPROVED** [review]

**Comment by nastra** [comment]
thanks for taking care of this @manuzhang

**Pull request merged**